File: lib/search_kit/match.rb
module SearchKit
  Match = OSX::Match
  
  class Match < OSX::NSObject
    kvc_accessor :name
    
    def name
      @name ||= RubyName.from_karidoc_path(URL().path)
    end
    objc_method :name, [:id]
    
    def filename
      OSX::NSURL.fileURLWithPath(File.join(Manager.current_filepath, URL().path))
    end
    
    def inspect
      "#<SearchKit::Match:#{object_id} name=#{name} url=#{URL().path} score=#{score}>"
    end
  end
end

File: lib/search_kit/match.m
#import "Match.h"

@implementation Match

-(NSURL*)URL { return _url; }
-(void)setURL:(NSURL*)url { _url = [url copy]; }

-(NSNumber*)score { return _score; }
-(void)setScore:(NSNumber*)score { _score = [score copy]; }

-(Match*)initWithURL:(NSURL*)url score:(NSNumber*)score {
  if ([super init]) {
    [self setURL:url];
    [self setScore:score];
    return self;
  } else {
    return nil;
  }
}

@end

void Init_Match() {}